### PR TITLE
Updating trace document

### PR DESCRIPTION
**Pull request by ZhuYuJin**
This PR is following the previous PR: #3598 

The PR updates the trace document with the new feature of tracing tensors.

### CODE REVIEW

**Comment by ZhuYuJin** [comment]
@deadeyegoodwin Please check the document update.

**Review by GuanLuo — DISMISSED** [review]

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: docs/trace.md [hunk 1]
```diff
@@ -67,18 +72,18 @@ The trace output is a JSON file with the following schema.
...
+"id" of the containing ensemble.  
+
+Each `TIMESTAMPS` trace will have one or more
 "timestamps" with each timestamp having a name and the timestamp in
 nanoseconds ("ns"). For example,
 
 ```
 [
   {
-    "model_name": "simple",
```
Comment (deadeyegoodwin) [inline docs/trace.md:80]: Why did you remove these lines. The trace file should contain these entries.

File: docs/trace.md [hunk 2]
```diff
@@ -95,6 +100,25 @@ nanoseconds ("ns"). For example,
...
 
+Each `TENSORS` trace will contain an "activity" and a "tensor". "activity" indicates 
+the type of tensor, including "TENSOR_QUEUE_INPUT" and "TENSOR_BACKEND_OUTPUT" by now.
+"tensor" has the detail of tensor, including its "name", "data" and "dtype". 
+For example,
+
+```
+[
+  {
+    "id":1,
```
Comment (deadeyegoodwin) [inline docs/trace.md:111]: This should also have the model name and version

File: docs/trace.md [hunk 3]
```diff
@@ -191,7 +259,7 @@ The meaning of the trace timestamps is:
 
   * Compute: The time the inference request spent executing the actual
```
Comment (deadeyegoodwin) [inline docs/trace.md:260]: Should be at same level as "Queue" above

File: docs/trace.md [hunk 4]
```diff
@@ -207,3 +275,14 @@ The meaning of the trace timestamps is:
 
   * Overhead: Additional time required for request handling not
```
Comment (deadeyegoodwin) [inline docs/trace.md:276]: Should be at same level as "Queue" and "Compute"

File: docs/trace.md [hunk 5]
```diff
@@ -207,3 +275,14 @@ The meaning of the trace timestamps is:
...
     covered by Queue or Compute times.
+
+* Data Flow: The data flow of the first request. It contains the input and 
+  output tensors of each part of execution.
+
+  * Name: The name of model.
```
Comment (deadeyegoodwin) [inline docs/trace.md:282]: Name and version should be at a higher level. What does a complete trace summary with TIMESTAMPS and TRACE look like?

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: docs/trace.md (hunk 3 above)
Comment (ZhuYuJin) [reply docs/trace.md:260]: "Compute" is the same level as "Queue". What's your meaning?

File: docs/trace.md (hunk 5 above)
Comment (ZhuYuJin) [reply docs/trace.md:282]: The "TIMESTAMPS" output is the same as before. The "TRACE" output is an extra display in the terminal. An example is shown in the lines from 204 to 238.
Each model entry has a "Name", "Version", "QUEUE_INPUT" and "BACKEND_OUTPUT". If we run an ensemble model, several model entries is shown hierarchically.

File: docs/trace.md [hunk 6]
```diff
@@ -29,24 +29,30 @@
...
 protocol](protocol/extension_trace.md).
 
+## Supported Trace Level Option
+
+- `TIMESTAMPS`: Tracing execution timestamps of each request.
+- `TENSORS`: Tracing input and output tensors during the execution. 
+
 ## JSON Trace Output
 
 The trace output is a JSON file with the following schema.
```
Comment (deadeyegoodwin) [inline docs/trace.md:58]: You need to update the below schema to show how TENSORS output fits into it.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

**Comment by ZhuYuJin** [comment]
@deadeyegoodwin Please check the latest commit.

**Review by deadeyegoodwin — APPROVED** [review]
